### PR TITLE
Add metric ontology

### PR DESCRIPTION
**Pull request by zhanyuanucb**
This PR fixes https://github.com/asyml/forte/issues/566

### Description of changes
Added `forte/ontology_specs/metric.json`

### Possible influences of this PR.
Due to the `ndarray` attribute in `NdMetric`, the metric ontology spec isn't ready to be auto-generated by the current ontology generator yet.

### Test Conducted
Unit tests are to be added

### CODE REVIEW

**Comment by hunterhector** [comment]
Could you do the following to make sure the PR is up to standard?

1. Change the PR name to the content instead of a number
2. Make the branch up to date to the branch
3. Make sure the CI passes.

**Comment by hunterhector** [comment]
This PR is fixing the wrong issue link

File: forte/ontology_specs/metric.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ 
+    "name": "metric", 
+    "definitions": [ 
+      {
+        "entry_name": "ft.onto.metric.BaseMetric",
```
Comment (hunterhector) [inline forte/ontology_specs/metric.json:5]: I believe we can simply use `Metric`, just to make things simple

File: forte/ontology_specs/metric.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        "description": "A base metric entity, all metric entities should inherit from it.", 
+        "attributes": [ 
+          { 
+              "name": "metric_name", 
+              "type": "str" 
+          }
+        ]
+      },
+      { 
+        "entry_name": "ft.onto.metric.SingleMetric",
```
Comment (hunterhector) [inline forte/ontology_specs/metric.json:16]: We will define some metrics as examples in this file later (such as accuracy), how about we add them now?

**Review by hunterhector — COMMENTED** [review]
We normally also add the generated class to Forte repo so users don't have to re-generate these.

File: forte/ontology_specs/metric.json (hunk 2 above)
Comment (zhanyuanucb) [reply forte/ontology_specs/metric.json:16]: I thought by assigning proper `metric_name` (e.g. `accuracy` or `loss_value`) metrics like accuracy or loss value could be covered by SingleMetric.

Comment (hunterhector) [reply forte/ontology_specs/metric.json:16]: I see. Having a `metric_name` is not a bad idea, maybe we should do that.

Comment (zhanyuanucb) [reply forte/ontology_specs/metric.json:16]: The base `Metric` class already had this `metric_name`

Comment (hunterhector) [reply forte/ontology_specs/metric.json:16]: i see, now i understand

**Review by hunterhector — APPROVED** [review]

**Pull request merged**